coroner-client/src/main/java/backtraceio/coroner/CoronerClient.java

```diff
@@ -0,0 +1,69 @@
+package backtraceio.coroner;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import backtraceio.coroner.common.HttpClient;
+import backtraceio.coroner.query.CoronerQueries;
+import backtraceio.coroner.query.CoronerQueryFields;
+import backtraceio.coroner.response.CoronerApiResponse;
+import backtraceio.coroner.response.CoronerHttpException;
+import backtraceio.coroner.response.CoronerResponse;
+import backtraceio.coroner.response.CoronerResponseException;
+
+public class CoronerClient {
+    private final HttpClient coronerHttpClient;
+    private final CoronerQueries coronerQueries;
+    private final List<String> DEFAULT_ATTRIBUTES = Arrays.asList(
+            CoronerQueryFields.FOLD_CALLSTACK,
+            CoronerQueryFields.FOLD_GUID,
+            CoronerQueryFields.FOLD_CLASSIFIERS
+    );
+
+    public CoronerClient(final String apiUrl, final String coronerToken) {
+        this(new CoronerHttpClient(apiUrl, coronerToken));
+    }
+
+    public CoronerClient(HttpClient httpClient) {
+        this.coronerHttpClient = httpClient;
+        this.coronerQueries = new CoronerQueries();
+    }
+
+    public CoronerResponse rxIdFilter(final String rxId) throws CoronerResponseException, CoronerHttpException, IOException  {
+        return this.rxIdFilter(rxId, new ArrayList<>());
+    }
+
+    public CoronerResponse rxIdFilter(final String rxId, final List<String> customAttributes) throws CoronerResponseException, CoronerHttpException, IOException {
```

Comment:
CoronerClient is a facade which provides concrete functionality needed. Builder pattern is used in classes `CoronerFiltersBuilder` and partially in `CoronerQueryBuilder`. The current implementation provides all currently required functionality, we will extend the builder options of these classes in case of future needs.

Reply:
Let's make it clear - you can have one method instead of concrete functions and you will have the same - or actually better, you will have less code. 

In my opinion - we can easily change that, but I don't treat it as a blocker for this pull request.